src/utils/provideLocalStorageDataClass.ts

```diff
@@ -195,3 +192,47 @@ function orderItems(
     order.map(([_, ascOrDesc]) => ascOrDesc),
   )
 }
+
+function compare({
+  itemValue,
+  filterValue,
+  opCode,
+}: {
+  itemValue: unknown
+  filterValue: string | number | boolean | null
+  opCode: 'eq' | 'neq' | 'gt' | 'lt' | 'gte' | 'lte'
+}): boolean {
+  if (['gt', 'lt', 'gte', 'lte'].includes(opCode)) {
+    if (typeof itemValue !== typeof filterValue) {
+      throw new Error(
+        `Invalid comparison: Item value ${itemValue} (type: ${typeof itemValue}) and` +
+          ` filter value ${filterValue} (type: ${typeof filterValue}) must be of the same type.`,
+      )
+    }
+    if (
+      typeof itemValue !== 'number' ||
+      typeof filterValue !== 'number' ||
+      typeof itemValue !== 'string' ||
+      typeof filterValue !== 'string'
+    ) {
+      throw new Error(
+        `Invalid comparison: ${itemValue} and ${filterValue} must be numbers or strings.`,
+      )
+    }
+
+    if (opCode === 'gt') return itemValue > filterValue
+    if (opCode === 'lt') return itemValue < filterValue
+    if (opCode === 'gte') return itemValue >= filterValue
+    if (opCode === 'lte') return itemValue <= filterValue
+
+    throw new Error(`Invalid comparison operator: ${opCode}`)
```

Comment:
* [ ] FYI Unreachable code according to the outer if.

Reply:
This is more a safeguard, that this also stays that way.

Reply:
This is unclear. If we expect the code to not handle all cases, see https://github.com/Scrivito/scrivito-portal-app/pull/525#discussion_r1692980563 for a solution.
If we expect unknown input from the SDK, we should type the input as `string`.

Reply:
FYI: I adjusted the error message.

Reply:
I see. So yes, I'd drop the check and use TypeScript to make sure we handle all cases.